familias/serializers.py

```diff
@@ -0,0 +1,208 @@
+            created objects, the integrante must be created first
+            and passed as parameter to the created function.
+
+            The Integrante model has a two sepcialization which are
+            indicated in the dictionary data as 'alumno_integrante'
+            and 'tutor_integrante'. This means the Integrante is
+            either a Tutor or an Alumno.
+
+            This function removes that data from the validated_data
+            dictionary, creates the Integrante and then creates the
+            specialized instance of either Alumno or Tutor.
+
+            Notes
+            -----
+            save_foreign_relashionship does not create object on None
+            object. Therefore if one of the specilizations is not
+            indicated it is not created.
+        """
+        self.validated_data['familia'] = family
+
+        alumno = self.validated_data.pop('alumno_integrante')
+        tutor = self.validated_data.pop('tutor_integrante')
+
+        integrante = Integrante.objects.create(**self.validated_data)
+
+        save_foreign_relashionship([alumno], AlumnoSerializer, integrante)
+        save_foreign_relashionship([tutor], TutorSerializer, integrante)
+
+        return integrante
+
+
+class FamiliaSerializer(serializers.ModelSerializer):
+    """ Serializer to represent a .models.Familia instance
+        through a REST endpoint for the offline application
+        to submit information.
+    """
+    integrante_familia = IntegranteSerializer(many=True)
+    comentario_familia = ComentarioSerializer(many=True)
+
+    class Meta:
+        model = Familia
+        fields = (
+            'id',
+            'numero_hijos_diferentes_papas',
+            'explicacion_solvencia',
+            'estado_civil',
+            'localidad',
+            'comentario_familia',
+            'integrante_familia',)
+
+    def create(self):
+        """ This function overides the default behaviour for creating
+            an object through a serializer.
+
+            Django Rest Framework does not implement saving nested
+            serializers.
+
+            Creates a Family instance. There are objects related to
+            the family that depend upon the creation of a family
+            instance first. This are the Integrantes and Comentarios.
+
+            We first remove all Integrante and comentario from the
+            validated data. We create the family and then we create
+            this objects that require the family instance.
+        """
+        integrantes = self.validated_data.pop('integrante_familia')
+        comentarios = self.validated_data.pop('comentario_familia')
+
+        family_instance = Familia.objects.create(**self.validated_data)
+
+        save_foreign_relashionship(integrantes, IntegranteSerializer, family_instance)
+        save_foreign_relashionship(comentarios, ComentarioSerializer, family_instance)
+
+        return family_instance
+
+    def update(self):
```

Comment:
This method will not serve the purposes of the offline application, and the way data has been modeled, since all of the indicadores, depend on specific family members existing over time, in order to keep track, of the changes in their incomes, as well as changes in becas given to specific children. I understand this is the easiest way to do this, but it has the potential to be catastrophic. Changes should be made, so the objects of integrantes as well as comentario are updated through their respective update methods.